templates/puppet_signscript.rb.erb

```diff
@@ -0,0 +1,43 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+#
+
+require 'net/http'
+require 'net/https'
+require 'fileutils'
+require 'json'
+require 'yaml'
+
+csr = STDIN.read
+
+settings_file    = File.exist?('/etc/puppetlabs/puppet/foreman.yaml') ? '/etc/puppetlabs/puppet/foreman.yaml' : '/etc/puppet/foreman.yaml'
+SETTINGS         = YAML.load_file(settings_file)
+protocol         = "<%= scope.lookupvar("foreman_proxy::puppetca_listen_on") %>"
+port             = "<% if scope.lookupvar("foreman_proxy::puppetca_listen_on") == "https" %><%= scope.lookupvar("foreman_proxy::ssl_port") -%>
+                    <%- else %><%= scope.lookupvar("foreman_proxy::http_port") -%>
+                    <%- end %>"
+# e.g. https://hostname.localdomain.com:8443/puppet/ca/autosign
+uri              = URI.parse("#{protocol}://<%= @fqdn %>:#{port}/puppet/ca/autosign")
+res              = Net::HTTP.new(uri.host, uri.port)
+res.use_ssl      = true
```

Comment:
`uri.scheme == 'https'` is better

Reply:
I just noticed that if you were not to use SSL (which we support by the erb-stuff) this will break anyway, so I will put this in another erb-block.